test/e2e/operator.go

```diff
@@ -419,3 +420,131 @@ var _ = Describe("ARO Operator - MHC Deployment", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Image Reconciler", func() {
+	imageconfigFlag := "aro.imageconfig.enabled"
+	optionalRegistry := "quay.io"
+	ctx := context.Background()
+	timeout := 5 * time.Minute
+	var expectedAllowlist, expectedBlocklist []string
+
+	getLocationRegistries := func(acrDomain, acrDnsSuffix, location string) []string {
+		regionalAcr := fmt.Sprintf("arosvc.%s.data.%s", location, acrDnsSuffix)
+		requiredRegistries := []string{acrDomain, regionalAcr}
+		return requiredRegistries
+	}
+
+	sliceEqual := func(a, b []string) bool {
+		if len(a) != len(b) {
+			return false
+		}
+		sort.Strings(a)
+		sort.Strings(b)
+
+		for idx, entry := range b {
+			if a[idx] != entry {
+				return false
+			}
+		}
+		return true
+	}
+
+	verifyLists := func() (bool, error) {
```

Comment:
As it turns out `Eventually()` has [another way it can be used](https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode) to make assertions inside the function and I was able to leverage that to pass in the current state.